index.js

```diff
@@ -739,9 +704,16 @@ RedisClient.prototype.send_command = function (command, args, callback) {
 
     // if the value is undefined or null and command is set or setx, need not to send message to redis
     if (command === 'set' || command === 'setex') {
-        if(args[args.length - 1] === undefined || args[args.length - 1] === null) {
+        if (args.length === 0) {
+            return;
+        }
+        if (args[args.length - 1] === undefined || args[args.length - 1] === null) {
             var err = new Error('send_command: ' + command + ' value must not be undefined or null');
-            return callback && callback(err);
+            if (callback) {
+                return callback && callback(err);
+            }
+            this.emit("error", err);
```

Comment:
If an error is returned to the callback the same behavior should be applied if no callback is present. So this is going to emit an error in that case from now on.